frontend/src/pages/projects/screens/detail/ProjectDetailsComponents.tsx

```diff
@@ -28,7 +28,9 @@ const ProjectDetailsComponents: React.FC = () => {
     dataConnections: { data: connections, loaded: connectionsLoaded },
     servingRuntimes: { data: modelServers, loaded: modelServersLoaded },
   } = React.useContext(ProjectDetailsContext);
-  const { pipelinesServer } = usePipelinesAPI();
+
+  // Limit set to 1 for checking pipeline emptiness
+  const [pipelines, pipelinesLoaded] = usePipelines(1);
```

Comment:
This is weighty and will attempt a call when the server is not even set up. Not sure this is the best way to go about it. I would think we might want to leverage pushing the decision down to the component where the context has more value.

Reply:
Push it to the layer that knows more... separation of concerns. At this level the server is potentially not created, so you're just blindly fetching and I'm saying do more effort and put the logic in with the sections -- `isEmpty` was the work of @DaoDaoNoCode BEFORE Pipelines was added... the data is different, it's not scalable, it now needs to change. This is not going to be an easy refactor as the information isn't "just put it on layer x and it's done".

Feel free to start a thread on the Slack channel to discuss the technical needs with the team. I'm not sure a back and forth on GitHub PRs will be effective.